lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -644,13 +644,17 @@ class LLDBTypeInfoProvider : public swift::remote::TypeInfoProvider {
     if (auto ti = m_runtime.lookupClangTypeInfo(clang_type))
       return *ti;
 
+    auto &process = m_runtime.GetProcess();
+    ExecutionContext exe_ctx;
+    process.CalculateExecutionContext(exe_ctx);
+    auto *exe_scope = exe_ctx.GetBestExecutionContextScope();
     // Build a TypeInfo for the Clang type.
-    auto size = clang_type.GetByteSize(nullptr);
-    auto bit_align = clang_type.GetTypeBitAlign(nullptr);
+    auto size = clang_type.GetByteSize(exe_scope);
+    auto bit_align = clang_type.GetTypeBitAlign(exe_scope);
```

Comment:
What is GetNumFields using the scope for? I'm asking because this will produce a ExecutionContextScope for the entire process, and if we would get more accurate results if were to require passing in an exe_scope for a specific frame.
But if all the exe_scope is used for is to get at the process, then this is perfectly fine.

Reply:
`GetBitSize` only uses it to get back to the process, `GetTypeBitAlign` and `GetNumFields` apparently doesn't them at all (at least not in `TypeSystemClang`.